### PR TITLE
Add header/footer support to entities card

### PR DESCRIPTION
**Pull request by balloob**
image

```yaml
type: entities
header:
  type: picture
  image: 'https://le-reve.com/wp-content/uploads/Balloons_Banner.png'
footer:
  type: picture
  image: 'https://www.freeiconspng.com/uploads/picture-of-fire-flames--cliparts-22.png'
entities:
  - entity: light.bed_light_2
  - entity: light.ceiling_lights
  - entity: light.kitchen_lights_2
```

Add support to the entities card to have header/footer elements. The difference between a header/footer element and a first/last row is that there is no padding around the element. This allows for more flexibility on what is possible. 

For this first iteration I just did a quick picture header/footer. It supports taps as it's pretty much a copy of the hui-picture-card but without the `<ha-card>`

I created a new `create-header-footer-element.ts` file but really, we should start extracting this logic into a new reusable function. We keep copy pasting the same code 🤷‍♂

### CODE REVIEW

File: src/panels/lovelace/cards/hui-entities-card.ts [hunk 1]
```diff
@@ -130,12 +135,18 @@ class HuiEntitiesCard extends LitElement implements LovelaceCard {
...
+          ? this.renderHeaderFooter(this._config.footer, "footer")
+          : ""}
       </ha-card>
     `;
   }
 
   static get styles(): CSSResult {
     return css`
+      :host {
+        --ha-card-border-radius: 20px;
```
Comment (iantrich) [inline src/panels/lovelace/cards/hui-entities-card.ts]: Is this an error? We don't want this on just one card

Comment (balloob) [reply src/panels/lovelace/cards/hui-entities-card.ts]: oh shit that was my debugging :D 2px was too difficult to see if my patches for the heaer/footer class worked.

File: src/panels/lovelace/cards/hui-entities-card.ts [hunk 2]
```diff
@@ -19,6 +19,8 @@ import { processConfigEntities } from "../common/process-config-entities";
 import { createRowElement } from "../common/create-row-element";
 import { EntitiesCardConfig, EntitiesCardEntityConfig } from "./types";
 import { applyThemesOnElement } from "../../../common/dom/apply_themes_on_element";
+import { createHeaderFooterElement } from "../common/create-header-footer-element";
```
Comment (balloob) [inline src/panels/lovelace/cards/hui-entities-card.ts]: Do we want to have 1 method for both headers and footers? Will there be elements that really better fit just a header or footer ?

Comment (iantrich) [reply src/panels/lovelace/cards/hui-entities-card.ts]: aesthetically, probably, but realistically, I doubt it.

File: src/panels/lovelace/cards/hui-picture-card.ts [hunk 3]
```diff
@@ -79,18 +80,20 @@ export class HuiPictureCard extends LitElement implements LovelaceCard {
...
+
     return html`
       <ha-card
         @action=${this._handleAction}
         .actionHandler=${actionHandler({
           hasHold: hasAction(this._config!.hold_action),
           hasDoubleClick: hasAction(this._config!.double_tap_action),
         })}
-        tabindex="0"
+        tabindex=${ifDefined(clickable ? 0 : undefined)}
```
Comment (iantrich) [inline src/panels/lovelace/cards/hui-picture-card.ts]: hitting enter currently only supports tap_action, so should be the only one considered for setting tabindex.

File: src/panels/lovelace/cards/hui-picture-card.ts [hunk 4]
```diff
@@ -79,18 +80,20 @@ export class HuiPictureCard extends LitElement implements LovelaceCard {
       return html``;
     }
 
+    const clickable = Boolean(
+      this._config.tap_action || this._config.hold_action
```
Comment (iantrich) [inline src/panels/lovelace/cards/hui-picture-card.ts]:
```suggestion
      this._config.tap_action || this._config.hold_action || this._config.double_tap_action
```

Comment (iantrich) [reply src/panels/lovelace/cards/hui-picture-card.ts]: probably a `hasActions(this._config)` function would be best for re-usability by other elements

**Comment by bramkragten** [comment]
I wonder if we should just use the normal cards and send a property with it if it is a header/footer so it can drop the padding

File: src/panels/lovelace/cards/hui-picture-card.ts (hunk 3 above)
Comment (bramkragten) [reply src/panels/lovelace/cards/hui-picture-card.ts]: `hasAction(this._config.tap_action)`

File: src/panels/lovelace/cards/hui-picture-card.ts (hunk 4 above)
Comment (balloob) [reply src/panels/lovelace/cards/hui-picture-card.ts]: Right. Also I didn't see you did this in #4494 so I'll remove this again.

**Comment by balloob** [comment]
It gets more complicated. It's not just padding. It has to actually remove the whole card. And in the case of picture-card, we're now having to attach all handlers to the `` instead, ending up with a pretty complicated component. Better to make a new one. 

I do think that we should pass in to the component if it's used as header or footer. 

For cards, we can always extract the logic that renders the card into a new component and re-use that in header/footer if it's useful. We already extracted all the actions stuff so most logic is already reusable.

**Comment by iantrich** [comment]
extracting said logic would probably be useful for other things as well. e.g. entity-button card vs service-button element. conditional-row vs conditional-card (I should have done that in my PR and will do so in a follow-up)

**Comment by balloob** [comment]
Needs to be rebased once #4509 has been merged and adopted to use the new createElement function.

**Review by bramkragten — APPROVED** [review]

**Comment by iantrich** [comment]
documentation needed

**Comment by balloob** [comment]
https://github.com/home-assistant/home-assistant.io/pull/11824

**Pull request merged**